### PR TITLE
AQUA. Adds additional fields into the AquaModel class.

### PR DESCRIPTION
**Pull request by mrDzurb**
## Description
https://jira.oci.oraclecorp.com/browse/ODSC-56571

Adds additional fields into the  ```AquaModel``` class.  

- inference_container: str = None
- finetuning_container: str = None
- evaluation_container: str = None

**The fields will be populated in the requests:**

- aqua/model/hf/search
- aqua/model/get

### CODE REVIEW

**Review by darenr — APPROVED** [review]

**Review by mingkang111 — APPROVED** [review]

File: ads/aqua/model.py [hunk 1]
```diff
@@ -53,17 +53,24 @@
...
+    MetadataTaxonomyKeys,
+    ModelCustomMetadata,
+    ModelCustomMetadataItem,
+)
 from ads.telemetry import telemetry
 
-from ads.common.extended_enum import ExtendedEnum
+
+class ModelCustomMetadataFields(ExtendedEnum):
+    ARTIFACT_LOCATION = "artifact_location"
```
Comment (VipulMascarenhas) [inline ads/aqua/model.py:70]: no action needed: we have all these constants in config/constants/etc, but it's better to have in a enum like this.

Comment (mrDzurb) [reply ads/aqua/model.py:70]: Maybe we should move it somewhere else :) Looks like Ming is doing a big refactoring and sorting out the modules, constants, utils, e.t.c. Maybe  during this big refactoring this enum will be moved into proper place.

File: ads/aqua/model.py [hunk 2]
```diff
@@ -469,33 +479,48 @@ def get(self, model_id) -> "AquaModel":
...
-                    AQUA_SERVICE_MODELS_BUCKET, CONDA_BUCKET_NS, artifact_path
-                ).path
-        except ValueError:
-            artifact_path = utils.UNKNOWN
 
-        if not artifact_path:
-            logger.debug("Failed to get artifact path from custom metadata.")
+        model_card = ""
+        if load_model_card:
+            artifact_path = get_artifact_path(ds_model.custom_metadata_list)
```
Comment (VipulMascarenhas) [inline ads/aqua/model.py:485]: will list iteration work on ds_model.custom_metadata_list?  It's type is`ModelCustomMetadata` whereas `get_artifact_path` takes in a list.

Comment (mrDzurb) [reply ads/aqua/model.py]: Fixed

**Review by VipulMascarenhas — APPROVED** [review]

File: ads/aqua/model.py [hunk 3]
```diff
@@ -491,18 +522,14 @@ def get(self, model_id) -> "AquaModel":
         aqua_model_atttributes = dict(
```
Comment (lu-ohai) [inline ads/aqua/model.py:522]: Seems like there is a typo for `aqua_model_attributes`

**Pull request merged**